fedimintd/src/ui.rs

```diff
@@ -144,7 +144,17 @@ async fn post_guardians(
     let dir_out_path = state.cfg_path.clone();
     let fedimintd_sender = state.sender.clone();
 
+    // kill dkg if it's already running
+    if let Some(dkg_task_group) = state.dkg_task_group.clone() {
+        tracing::info!("killing dkg task group");
```

Comment:
It's not really killing, just graceful shutting-down. It's not supposed to trigger any errors etc if the code is written right.